peekaboo/toolbox/ms_office.py

```diff
@@ -59,3 +59,34 @@ def has_office_macros(office_file):
     except Exception as error:
         logger.exception(error)
         return False
+
+def has_office_macros_with_auto_action(office_file):
+    """
+    Detects macros for Auto_ actions in Microsoft Office documents.
+
+    @param office_file: The MS Office document to check for Auto_ macros.
+    @return: True if Auto_ macros where found, otherwise False.
+             If VBA_Parser crashes it returns False too.
+    """
+
+    SUSPICIOUS_KEYWORDS = ["AutoOpen", "AutoClose"]
+
+    file_extension = office_file.split('.')[-1]
+    if file_extension not in MS_OFFICE_EXTENSIONS:
+        print("/%s/" % file_extension)
+        return False
+    try:
+        # VBA_Parser reports macros for office documents
+        vbaparser = VBA_Parser(office_file)
```

Comment:
The docs (https://github.com/decalage2/oletools/wiki/olevba#close-the-vba_parser) say, we should explicitly close the parser after use, to make sure the file handle is closed. This might be a good idea for both functions just to be on the safe side. We quite likely currently get the same just from garbage collection but this still leaves room for file handle exhaustion by delayed cleanup or internal changes to oletools in the future making explicit close non-optional.

Reply:
Fixed in 6a3ba2803d3f077b836d9755fb46e273bdb912b0.